x/concentrated-liquidity/model/pool_test.go

```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
+// TestGetAddress tests the GetAddress method of pool
+func (s *ConcentratedPoolTestSuite) TestGetAddressAndGetIncentiveAddress() {
+
+	tests := []struct {
+		name          string
+		expectedPanic bool
+	}{
+		{
+			name: "Happy path",
+		},
+		{
+			name:          "Unhappy path: wrong bech32 encoded address",
+			expectedPanic: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a concentrated liquidity pool struct instance
+			address := s.TestAccs[0].String()
+
+			// if the test case is expected to panic, we use wrong bech32 encoded address instead
+			if tc.expectedPanic {
+				address = "osmo15l7yueqf3tx4cvpt6njvj7zxmvuhkwyrr509e9"
+			}
+			mock_pool := model.Pool{
+				Id:      1,
+				Address: address,
+			}
+
+			// Check that the returned address is backward compatible
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				addr := mock_pool.GetAddress()
+				s.Require().Equal(addr, s.TestAccs[0])
+			})
+		})
+	}
+}
+
+// TestGetIncentivesAddress tests the GetIncentivesAddress method of pool
+func (s *ConcentratedPoolTestSuite) TestGetIncentivesAddress() {
+
+	tests := []struct {
+		name          string
+		expectedPanic bool
+	}{
+		{
+			name: "Happy path",
+		},
+		{
+			name:          "Unhappy path: wrong bech32 encoded address",
+			expectedPanic: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a concentrated liquidity pool struct instance
+			address := s.TestAccs[0].String()
+
+			// if the test case is expected to panic, we use wrong bech32 encoded address instead
+			if tc.expectedPanic {
+				address = "osmo15l7yueqf3tx4cvpt6njvj7zxmvuhkwyrr509e9"
+			}
+			mock_pool := model.Pool{
+				Id:      1,
+				Address: address,
+			}
+
+			// Check that the returned address is backward compatible
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				addr := mock_pool.GetAddress()
+				s.Require().Equal(addr, s.TestAccs[0])
+			})
+		})
+	}
+}
+
+// TestString tests if String method of the pool correctly json marshals the pool object
+func (s *ConcentratedPoolTestSuite) TestString() {
+	s.Setup()
+
+	pool, err := model.NewConcentratedLiquidityPool(1, "foo", "bar", DefaultTickSpacing, DefaultSpreadFactor)
+	s.Require().NoError(err)
+	poolString := pool.String()
+	s.Require().Equal(poolString, "{\"address\":\"osmo19e2mf7cywkv7zaug6nk5f87d07fxrdgrladvymh2gwv5crvm3vnsuewhh7\",\"incentives_address\":\"osmo156gncm3w2hdvuxxaejue8nejxgdgsrvdf7jftntuhxnaarhxcuas4ywjxf\",\"fees_address\":\"osmo1nsz03mrk6krk5p86f4qr48kden2pt39afw35qzvgqp8m09jwafzqdl8nsk\",\"id\":1,\"current_tick_liquidity\":\"0.000000000000000000\",\"token0\":\"foo\",\"token1\":\"bar\",\"current_sqrt_price\":\"0.000000000000000000\",\"tick_spacing\":1,\"exponent_at_price_one\":-6,\"spread_factor\":\"0.010000000000000000\",\"last_liquidity_update\":\"0001-01-01T00:00:00Z\"}")
```

Comment:
Is there any way we can Printf some of these items to match the defaults as well as what was provided? I would like it so that if we change defaultTickSpacing or DefaultSpreadFactor that we wouldn't need to come back and modify these as well

Reply:
It would technically be possible ( by referencing the proto file and it's json tag), but I definitely don't think its worth the struggle to automate this